### PR TITLE
Add print-method implementations

### PR DESCRIPTION
**Pull request by patrickgombert**
* Keyword
  * Symbol
  * ISeq
  * IPersistentMap

### CODE REVIEW

File: test/clj/clojure/lang/out_test.clj [hunk 1]
```diff
@@ -14,13 +15,51 @@
...
+
+  (testing "pr for a symbol"
+    (is (=
+          (with-out-str (pr (symbol "test")))
+          "test")))
+
+  (testing "pr for a seq"
+    (is (re-matches
+          #".*TestSeqable.*"
+          (with-out-str (pr (test-seqable '(1 2 3)))))))
```
Comment (mylesmegyesi) [inline test/clj/clojure/lang/out_test.clj]: Any reason we can't assert against the whole string?

File: test/clj/clojure/lang/out_test.clj [hunk 2]
```diff
@@ -14,13 +15,51 @@
...
+
+  (testing "pr for a seq with meta"
+    (binding [*print-meta* true
+              *print-readably* true]
+      (let [seq-with-meta (reify
+                            ISeqable
+                            (-seq [this] '(1 2 3))
+                            IMeta
+                            (-meta [this] (array-map :so :meta)))]
+        (is (= "^{:so :meta}"
```
Comment (mylesmegyesi) [inline test/clj/clojure/lang/out_test.clj]: ibid

File: src/clj/clojure/next.clj [hunk 3]
```diff
@@ -225,7 +225,9 @@
   (is-zero? i))
 
 (defn count [obj]
-  (-count obj))
+  (if obj
+    (-count obj)
+    0))
```
Comment (mylesmegyesi) [inline src/clj/clojure/next.clj]: I'd like to see a test for this.

File: src/clj/clojure/next.clj [hunk 4]
```diff
@@ -792,18 +797,89 @@
...
+      (if (and (= (count m) 1) (:tag m))
+          (print-method (:tag m) w)
+          (print-method m w))
+      (platform-write w " "))))
+
+(defn print-simple [obj wrtr]
+  (print-meta obj wrtr)
+  (platform-write wrtr (str obj)))
+
+(defn- print-sequential [begin, print-one, sep, end, sequence, w]
```
Comment (mylesmegyesi) [inline src/clj/clojure/next.clj]: commas?

File: src/clj/clojure/next.clj [hunk 5]
```diff
@@ -792,18 +797,89 @@
...
+            ; TODO: switch to [[x & xs] xs] once
+            ; it is using clojure.core/nth
+            (loop [rxs xs]
+              (let [x (first rxs)
+                    xs (next rxs)]
+                (print-one x w)
+                (when xs
+                  (platform-write w sep)
+                  (recur xs))))))
+        (platform-write w end)))))
```
Comment (mylesmegyesi) [inline src/clj/clojure/next.clj]: This method is huge. I understand why, but I'd like to see a lot more tests around this behavior. I count at least 6 `if` statements and only two tests.

File: test/clj/clojure/lang/out_test.clj (hunk 1 above)
Comment (patrickgombert) [inline test/clj/clojure/lang/out_test.clj]: It has the @hashcode thing going on and the hashcode isn't consistent. I'm realizing that I wanted a `test-seq` not `not-seqable` here though.

File: src/clj/clojure/next.clj (hunk 4 above)
Comment (patrickgombert) [inline src/clj/clojure/next.clj]: copypasta

File: src/clj/clojure/next.clj (hunk 3 above)
Comment (patrickgombert) [inline src/clj/clojure/next.clj]: :+1:

**Comment by patrickgombert** [comment]
updated

**Pull request merged**